### PR TITLE
changed Byte to Bool

### LINKED ISSUE #49769: torch.where should mention Bool rather than Byte when given wrong dtype mask
## 🐛 Bug

image

 - PyTorch Version (e.g., 1.0): 1.7.0

### PR DESCRIPTION
**Pull request by jonykarki**
Fixes #49769

### CODE REVIEW

File: aten/src/ATen/native/TensorCompare.cpp [hunk 1]
```diff
@@ -217,7 +217,7 @@ Tensor where(const Tensor& condition, const Tensor& self, const Tensor& other) {
               condition.device(), " and x and y are on ", self.device(), " and ", other.device(),
               " respectively");
   TORCH_CHECK(condition.scalar_type() == ScalarType::Byte || condition.scalar_type() == ScalarType::Bool,
-              "Expected condition to have ScalarType Byte, but got ScalarType ",
```
Comment (mruberry) [inline aten/src/ATen/native/TensorCompare.cpp:220]: Let's tweak this a bit

```
if (condition.scalar_type() == ScalarType::Byte) {
  TORCH_WARN_ONCE("where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead.");
} else {
  TORCH_CHECK(condition.scalar_type() == ScalarType::Bool, "where expected condition to be a boolean tensor, but got a tensor with dtype ", condition.scalar_type());
}
```

That updates the error's language and correctly warns when the condition is a byte tensor.
